Example/AsyncReactorExample/Features/Repository/Search/RepositorySearchReactor.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  TestReactor.swift
+//  Rudi
+//
+//  Created by Dominik Arnhof on 14.04.23.
+//
+
+import Foundation
+import AsyncReactor
+import UIKit
+import Logging
+
+private let logger = Logger(label: "RepositorySearchReactor")
+
+class RepositorySearchReactor: AsyncReactor {
+    enum Action {
+        case onHidePrivateToggle
+        case enterQuery(String)
+        case load
+    }
+    
+    struct State {
+        var hidePrivate = false
+        var query = ""
+        var repositories: [Repository] = []
+        var isLoading = false
+    }
+    
+    @Published
+    private(set) var state: State
+    
+    @MainActor
+    init(state: State = State()) {
+        self.state = state
+        
+        lifecycleTask {
```

Comment:
Maybe you can find an actual example use case for the `lifecycleTask`? This shows how to use it, but does not really make sense to toggle the switch when the application becomes active.